### ISSUE TITLE
Night Vision Mode 🌚

### ISSUE THREAD

**Issue by MohamedAtef321** [ultralytics/ultralytics#1568]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello, 

- I want to enhance the results in dark environments.
- That's why I created a feature to do so and I am asking for a PR.

Here's an example of the result before and after Night Vision feature: \

![image](https://user-images.githubusercontent.com/73308647/226955672-05ed1fef-be83-4cd4-a6f6-e3c6ee309255.png)

Please consider my work, and have a look on it ✅.

Thanks

### Use case

Enhance the results in dark environments.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by MohamedAtef321** [comment]
You shouldn't pass it like `night_vision=True`, instead you should pass it like `night_vision` or `night_vision=show` to see results, I have made a little documentation for this feature usage in the `docs/tasks/detect.md` file.

And here are the little documentation I am talking about :
 
------------------------------------------------------

## Night Vision Feature
If you use camera in a dark environment, then you may get poor results.\
We have made a preprocess feature to enhance image brightness, it actually may help you get better results in dark environments.

You can use Night Vision mode by adding parameter `night_vision` as follows:

Using CLI:
```bash
yolo detect predict model=path/to/best.pt source='https://ultralytics.com/images/bus.jpg' night_vision
```
and it has 3 modes:

1- Default mode:
> By not passing parameter `night_vision`, it will use the original image as it is.

2- Night Vision mode (Only apply on input image):
```bash
night_vision 
```
> It will apply night vision on input image, pass it to the model and get the result based on night-processed image, But the shown/saved image will be the original image without night filter.

3- Night Vision mode (Apply on both input image and shown/saved image):
```bash
night_vision=show
```
> It will apply night vision on input image, pass it to the model and get the result based on night-processed image, The shown/saved image will be also night-processed image with night filter applied.

An example to differentiate between Night Vision modes and their (Saved/Shown) results:

<img width="716" alt="night_vision_example" src="https://user-images.githubusercontent.com/73308647/227236156-e51d72ea-75ce-4eb4-b3ec-700ddbc4bc8c.png">

**Comment by MohamedAtef321** [comment]
I am going to check it for a video for you, and I will show you the results. 👍

**Comment by pderrenger** [comment]
@MohamedAtef321 hi there! 👋 Thank you for your enthusiasm in enhancing YOLOv8 for dark environments! YOLOv8 currently supports different modes like Train, Val, Predict, Export, Track, and Benchmark, but incorporating a Night Vision feature for improving results in dark environments is an interesting idea. 

However, for image enhancement like night vision, we typically recommend using image pre-processing techniques to adjust brightness, contrast, or apply specific filters before passing the images through the YOLOv8 model during prediction. This ensures that YOLOv8 focuses on its core strength of object detection while allowing for flexible image pre-processing tailored to different scenarios.

Feel free to give it a go and let us know if you encounter any issues during implementation. It will be great to see your work as a PR! 🚀

If you need any further guidance on integrating image pre-processing with YOLOv8, feel free to ask. We're here to help!

**Comment by MohamedAtef321** [comment]
@pderrenger Thanks for your great words about our idea.🙏 
I agree with you on this. that's why when we worked on adding new features for dark environments, we added them as optional features in the pre-processing section of YOLOv8. We didn't want to affect YOLOv8 core, we instead wanted to make it more extended to deal with different situations without affecting its performance.
We intend to add more optional customized image-processing functions to your YOLOv8 pre-processing. At the current time, we are more concerned with "Job seeking" as you know, we want our life to be more stable.😅
Once we have time, we will be very happy to contribute to your YOLOv8. It's an exciting thing to do.⚡
